src/mono/wasm/runtime/jiterpreter-jit-call.ts

```diff
@@ -189,9 +189,28 @@ export function mono_interp_invoke_wasm_jit_call_trampoline(
     const thunk = <Function>getWasmTableEntry(thunkIndex);
     try {
         thunk(ret_sp, sp, ftndesc, thrown);
-    } catch (exc) {
+    } catch (exc: any) {
         receiveWorkerHeapViews();
-        setU32_unchecked(thrown, 1);
+        const exceptionTag = (<any>Module)["asm"]["__cpp_exception"];
+        const haveTag = exceptionTag instanceof (<any>WebAssembly).Tag;
+        if (
+            !haveTag || (
+                (exc instanceof (<any>WebAssembly).Exception) &&
+                exc.is(exceptionTag)
+            )
+        ) {
+            setU32_unchecked(thrown, 1);
+
+            // Wasm EH is enabled, so we know that the current exception is a C++ exception
+            // Call begin_catch and then end_catch to clean it up.
+            if (haveTag) {
```

Comment:
should there be else branch for `haveTag` ?

Reply:
If we don't have a tag I think that means we're dealing with emscripten's JS-based exception handling. We would probably try to interface it inside an else arm there.

Reply:
at least re-throw ?

Reply:
that would make it impossible to handle exceptions of any kind, which would break everything instead of just Environment.Exit. I probably need to figure out how to detect emscripten js-mode exceptions before merging this I guess...